include/gatekeeper_lls.h

```diff
@@ -19,6 +19,69 @@
 #ifndef _GATEKEEPER_LLS_H_
 #define _GATEKEEPER_LLS_H_
 
-int run_lls(void);
+#include <rte_timer.h>
+
+#include "gatekeeper_mailbox.h"
+
+struct lls_cache {
+	/* Length (in bytes) of keys for this cache. */
+	uint32_t               key_len;
+
+	/* Timeout value (in seconds) to mark entries as stale. */
+	uint32_t               front_timeout_sec;
+	uint32_t               back_timeout_sec;
+
+	/* Name string needed for cache hash and mempools. */
+	const char             *name;
+
+	/* Array of cache entries indexed using @hash. */
+	struct lls_cache_entry **data;
+
+	/* Hash instance that maps IP address keys to LLS cache entries. */
+	struct rte_hash        *hash;
+
+	/* Memory pool for cache entries. */
+	struct rte_mempool     *entry_pool;
+
+	/* Mailbox to hold requests for changing the cache. */
+	struct mailbox         mb;
+
+	/*
+	 * Function to transmit a request out of @port_id to resolve
+	 * IP address @ip_key to an Ethernet address. If @ip_key
+	 * was previously known to map to @ha, then unicast a
+	 * request to that address; otherwise, @ha is NULL and the
+	 * request should be broadcast.
+	 */
+	void (*xmit_req)(uint8_t port_id, const void *ip_key,
+		const struct ether_addr *ha);
+
+	/* Function to print the cache entry that maps @ip_key to @entry. */
+	void (*print_entry)(struct lls_cache *cache, const void *ip_key,
+		const struct lls_cache_entry *entry);
+};
+
+struct lls_config {
+	unsigned int      lcore_id;
+	int               back_iface_enabled;
```

Comment:
Yes, thanks. While considering adding it to `struct net_config`, it occurred to me that our code currently assumes that there is a back interface, which is not the case for Grantor, for example. So I broke this addition of `back_iface_enabled` into a separate commit, so that if a back interface is not present, the code does not try to initialize it. Also, blocks that depend on a back interface will now throw an error.